server/svix-server/src/worker.rs

```diff
@@ -532,6 +520,24 @@ async fn dispatch(
     Ok(())
 }
 
+fn filter_expired_keys(
```

Comment:
It doesn't really filter, it combines the key with old key and then filters.

Have you considered just adding a function on the CreateMessageEndpoint struct that's called `valid_signing_keys()` or something like that which just gets it? Feels like something that's specific to that struct anyway.

Reply:
I took this approach because I could unit-test the functionality without having to instantiate an endpoint. But yeah, this would probably be better off as a method of CreateMessageEndpoint. One moment I'll have it wired up that way.